src/test/java/io/kestra/storage/s3/S3StorageTest.java

```diff
@@ -61,4 +63,19 @@ void getEmptyFile() throws IOException {
         assertThat(inputStream, not(instanceOf(ResponseInputStream.class)));
         assertThat(new BufferedReader(new InputStreamReader(inputStream)).lines().count(), is(0L));
     }
+
+    @Test
+    void mkdirRecursive() throws IOException {
+        storageInterface.createDirectory(null, URI.create("/first/second/third"));
+
+        List<FileAttributes> list = storageInterface.list(null, null);
+        assertThat(list, contains(
+            hasProperty("fileName", is("/"))
+        ));
+
+        list = storageInterface.list(null, URI.create("/"));
+        assertThat(list, contains(
+            hasProperty("fileName", is("first"))
+        ));
+    }
```

Comment:
It may be a good idea to add this into the StorageTestSuite and test this on all storage implementation